scripts/data_overlap/compute_data_overlap_metrics.py

```diff
@@ -296,50 +224,44 @@ def compute_scenario_document_data_overlap(
     hlog(f"Loading scenario data from {args.scenario_data}")
     light_scenarios = load_light_scenarios_from_jsonl(args.scenario_data)
 
+    stats_key_counts: DefaultDict[DataOverlapStatsKey, int] = defaultdict(int)
     with htrack_block("Initializing the stats, ngram_index, and ngram_counter"):
-        all_overlap_stats: AllDataOverlapStats
         ngram_index: NgramIndex
-        all_overlap_stats = create_all_data_overlap_stats(light_scenarios=light_scenarios, n_values=N_VALUES)
-        ngram_index = create_ngram_index(light_scenarios=light_scenarios, n_values=N_VALUES, tokenizer=tokenizer)
-        ngram_counter: NgramCounter = {}
+        ngram_index = create_ngram_index(
+            light_scenarios=light_scenarios, n_values=N_VALUES, tokenizer=tokenizer, stats_key_counts=stats_key_counts
+        )
+
+    # DataOverlapStatsKey -> Set[str] for ids
+    stats_key_to_input_ids: DefaultDict[DataOverlapStatsKey, Set] = defaultdict(set)
+    stats_key_to_reference_ids: DefaultDict[DataOverlapStatsKey, Set] = defaultdict(set)
```

Comment:
`defaultdict` is fine but you need to be careful about not accidentally adding unneeded entries when accessing the dict..